lib/puppet/provider/file/posix.rb

```diff
@@ -12,8 +12,19 @@
   require 'etc'
   require_relative '../../../puppet/util/selinux'
 
-  def self.post_resource_eval
-    Selinux.matchpathcon_fini if Puppet::Util::SELinux.selinux_support?
+  class << self
+    def selinux_handle
+      return nil unless Puppet::Util::SELinux.selinux_support?
+
+      @selinux_handle ||= Selinux.selabel_open(0, nil, 0)
```

Comment:
I think the first parameter needs to be whatever the `SELABEL_CTX_FILE` constant is defined to be